--- v8/SConstruct_origin	2012-04-10 11:21:39.000000000 -0500
+++ v8/SConstruct	2012-04-10 11:21:42.000000000 -0500
@@ -302,7 +302,7 @@
   'gcc': {
     'all': {
       'WARNINGFLAGS': ['-Wall',
-                       '-Werror',
+                       #'-Werror',
                        '-W',
                        '-Wno-unused-parameter',
                        '-Woverloaded-virtual',
@@ -366,7 +366,8 @@
 MKSNAPSHOT_EXTRA_FLAGS = {
   'gcc': {
     'os:linux': {
-      'LIBS': ['pthread'],
+      'LIBS': ['log']
+      #'LIBS': ['pthread'],
     },
     'os:macos': {
       'LIBS': ['pthread'],
