unchanged:
--- v8_orig/src/platform-linux.cc	2011-09-09 19:48:25.536842403 -0500
+++ v8/src/platform-linux.cc	2011-09-09 14:05:42.845135548 -0500
@@ -1085,7 +1085,7 @@
     if (result != 0 && errno != EINTR) {
       fprintf(stderr,
               "SignalSender usleep error; interval = %u, errno = %d\n",
-              interval,
+              (unsigned int)interval,
               errno);
       ASSERT(result == 0 || errno == EINTR);
     }
only in patch2:
unchanged:
--- v8_orig/src/platform-posix.cc	2011-09-09 19:48:25.540842403 -0500
+++ v8/src/platform-posix.cc	2011-09-09 19:55:02.772842683 -0500
@@ -48,7 +48,8 @@
 
 #if defined(ANDROID)
 #define LOG_TAG "v8"
-#include <utils/Log.h>  // LOG_PRI_VA
+#include <android/log.h>
+#define LOG_PRI_VA __android_log_print
 #endif
 
 #include "v8.h"
