From ebf43f2621cd22f1bb24d61c5da425dcfc53424a Mon Sep 17 00:00:00 2001
From: sgtcoolguy <chris.a.williams@gmail.com>
Date: Fri, 23 Oct 2015 23:44:16 -0400
Subject: [PATCH] Create standalone static libs

---
 icu.gyp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/v8/third_party/icu/icu.gyp b/v8/third_party/icu/icu.gyp
index ee206fe..6231e82 100644
--- a/v8/third_party/icu/icu.gyp
+++ b/v8/third_party/icu/icu.gyp
@@ -103,6 +103,7 @@
         {
           'target_name': 'icudata',
           'type': 'static_library',
+          'standalone_static_library': 1,
           'defines': [
             'U_HIDE_DATA_SYMBOL',
           ],
@@ -176,6 +177,11 @@
         {
           'target_name': 'icui18n',
           'type': '<(component)',
+          'conditions': [
+            [ 'component!="shared_library"', {
+                'standalone_static_library': 1,
+            }],
+          ],
           'sources': [
             '<@(icui18n_sources)',
           ],
@@ -257,6 +263,11 @@
         {
           'target_name': 'icuuc',
           'type': '<(component)',
+          'conditions': [
+            [ 'component!="shared_library"', {
+                'standalone_static_library': 1,
+            }],
+          ],
           'sources': [
             '<@(icuuc_sources)',
           ],
--
2.6.2
