diff --git a/v8/third_party/icu/icu.gyp b/v8/third_party/icu/icu.gyp
index ee206fe..d47fdaa 100644
--- a/v8/third_party/icu/icu.gyp
+++ b/v8/third_party/icu/icu.gyp
@@ -103,6 +103,7 @@
         {
           'target_name': 'icudata',
           'type': 'static_library',
+          'standalone_static_library': 1,
           'defines': [
             'U_HIDE_DATA_SYMBOL',
           ],
@@ -220,6 +221,9 @@
             },
           },
           'conditions': [
+            [ 'component!="shared_library"', {
+              'standalone_static_library': 1,
+            }],
             [ 'use_system_icu==1 and want_separate_host_toolset==1', {
               'toolsets': ['host'],
             }],
@@ -332,6 +336,9 @@
             },
           },
           'conditions': [
+            [ 'component!="shared_library"', {
+              'standalone_static_library': 1,
+            }],
             [ 'use_system_icu==1 and want_separate_host_toolset==1', {
               'toolsets': ['host'],
             }],
