diff --git a/gni/v8.gni b/gni/v8.gni
index a334651797..410e7d41d3 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -147,13 +147,7 @@ if (is_debug && !v8_optimized_debug) {
 } else {
   v8_remove_configs += [ "//build/config/compiler:default_optimization" ]
 
-  # TODO(crbug.com/621335) Rework this so that we don't have the confusion
-  # between "optimize_speed" and "optimize_max".
-  if (((is_posix && !is_android) || is_fuchsia) && !using_sanitizer) {
-    v8_add_configs += [ "//build/config/compiler:optimize_speed" ]
-  } else {
-    v8_add_configs += [ "//build/config/compiler:optimize_max" ]
-  }
+  v8_add_configs += [ "//build/config/compiler:optimize_speed" ]
 }

 if (!is_debug) {
