diff --git a/gni/v8.gni b/gni/v8.gni
index a334651797..410e7d41d3 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -147,13 +147,7 @@ if (is_debug && !v8_optimized_debug) {
 } else {
   v8_remove_configs += [ "//build/config/compiler:default_optimization" ]
 
-  # TODO(crbug.com/621335) Rework this so that we don't have the confusion
-  # between "optimize_speed" and "optimize_max".
-  if (((is_posix && !is_android) || is_fuchsia) && !using_sanitizer) {
-    v8_add_configs += [ "//build/config/compiler:optimize_speed" ]
-  } else {
-    v8_add_configs += [ "//build/config/compiler:optimize_max" ]
-  }
+  v8_add_configs += [ "//build/config/compiler:optimize_speed" ]
 }
 
 if (!is_debug) {
diff --git a/include/v8-version.h b/include/v8-version.h
index 8b88700b5c..760d2631da 100644
--- a/include/v8-version.h
+++ b/include/v8-version.h
@@ -11,7 +11,7 @@
 #define V8_MAJOR_VERSION 9
 #define V8_MINOR_VERSION 4
 #define V8_BUILD_NUMBER 146
-#define V8_PATCH_LEVEL 16
+#define V8_PATCH_LEVEL 0
 
 // Use 1 for candidates and 0 otherwise.
 // (Boolean macro values are not supported by all preprocessors.)
