diff --git a/build/android/BUILD.gn b/build/android/BUILD.gn
index 1234567..abcdefg 100644
--- a/build/android/BUILD.gn
+++ b/build/android/BUILD.gn
@@ -51,8 +51,6 @@ python_library("devil_chromium_py") {
   data = [
     "devil_chromium.py",
     "devil_chromium.json",
-    "//third_party/catapult/third_party/gsutil/",
-    "//third_party/catapult/devil/devil/devil_dependencies.json",
 
     # Read by gn_helpers.BuildWithChromium()
     "//build/config/gclient_args.gni",
@@ -123,7 +121,6 @@ python_library("resource_sizes_py") {
   pydeps_file = "resource_sizes.pydeps"
   data_deps = [
     ":devil_chromium_py",
-    "//third_party/catapult/tracing:convert_chart_json",
   ]
   data = [
     build_vars_file,
