From 27370c2180655ffc0423516e10333d93ed28cb88 Mon Sep 17 00:00:00 2001
From: Gary Mathews <contact@garymathews.com>
Date: Wed, 15 Sep 2021 12:41:41 -0700
Subject: [PATCH] compat_args

---
 include/v8.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/v8.h b/include/v8.h
index 154c03aa5b..0bf4f40e15 100644
--- a/include/v8.h
+++ b/include/v8.h
@@ -11536,7 +11536,7 @@ template<typename T>
 Local<Value> FunctionCallbackInfo<T>::operator[](int i) const {
   // values_ points to the first argument (not the receiver).
   if (i < 0 || length_ <= i) return Local<Value>(*Undefined(GetIsolate()));
-  return Local<Value>(reinterpret_cast<Value*>(values_ + i));
+  return Local<Value>(reinterpret_cast<Value*>(values_ - i));
 }
 
 template<typename T>
@@ -11547,7 +11547,7 @@ Local<Function> FunctionCallbackInfo<T>::Callee() const {
 template<typename T>
 Local<Object> FunctionCallbackInfo<T>::This() const {
   // values_ points to the first argument (not the receiver).
-  return Local<Object>(reinterpret_cast<Object*>(values_ - 1));
+  return Local<Object>(reinterpret_cast<Object*>(values_ + 1));
 }
 
 
-- 
2.25.1

