diff --git a/DEPS b/DEPS
index 439f45c..372d78e 100644
--- a/DEPS
+++ b/DEPS
@@ -88,6 +88,7 @@ vars = {
   'android_sdk_cmdline-tools_version': 'ZT3JmI6GMG4YVcZ1OtECRVMOLLJAWAdPbi-OclubJLMC',
 }
 
+# Note: android_ndk, android_platform, and android_sdk entries removed upstream
 deps = {
   'base/trace_event/common':
     Var('chromium_url') + '/chromium/src/base/trace_event/common.git' + '@' + '3da1e2fcf66acd5c7194497b4285ac163f32e239',
