diff --git a/DEPS b/DEPS
index 7ab0b3fc85..1499612827 100644
--- a/DEPS
+++ b/DEPS
@@ -24,14 +24,6 @@ deps = {
     Var('chromium_url') + '/chromium/buildtools.git' + '@' + '2888931260f2a32bc583f005bd807a561b2fa6af',
   'v8/base/trace_event/common':
     Var('chromium_url') + '/chromium/src/base/trace_event/common.git' + '@' + '0e9a47d74970bee1bbfc063c47215406f8918699',
-  'v8/third_party/android_ndk': {
-    'url': Var('chromium_url') + '/android_ndk.git' + '@' + 'e951c37287c7d8cd915bf8d4149fd4a06d808b55',
-    'condition': 'checkout_android',
-  },
-  'v8/third_party/android_tools': {
-    'url': Var('chromium_url') + '/android_tools.git' + '@' + '9a70d48fcdd68cd0e7e968f342bd767ee6323bd1',
-    'condition': 'checkout_android',
-  },
   'v8/third_party/catapult': {
     'url': Var('chromium_url') + '/catapult.git' + '@' + '8a42ad3cb185e340c32b20f657980fd057e3769f',
     'condition': 'checkout_android',
@@ -68,7 +60,6 @@ deps = {
 
 recursedeps = [
   'v8/buildtools',
-  'v8/third_party/android_tools',
 ]
 
 include_rules = [
