diff --git a/DEPS b/DEPS
index 1c3ffce539..530d101a23 100644
--- a/DEPS
+++ b/DEPS
@@ -26,14 +26,6 @@ deps = {
     Var('chromium_url') + '/chromium/buildtools.git' + '@' + '9a90d9aaadeb5e04327ed05775f45132e4b3523f',
   'v8/base/trace_event/common':
     Var('chromium_url') + '/chromium/src/base/trace_event/common.git' + '@' + '211b3ed9d0481b4caddbee1322321b86a483ca1f',
-  'v8/third_party/android_ndk': {
-    'url': Var('chromium_url') + '/android_ndk.git' + '@' + '5cd86312e794bdf542a3685c6f10cbb96072990b',
-    'condition': 'checkout_android',
-  },
-  'v8/third_party/android_tools': {
-    'url': Var('chromium_url') + '/android_tools.git' + '@' + '130499e25286f4d56acafa252fee09f3cc595c49',
-    'condition': 'checkout_android',
-  },
   'v8/third_party/catapult': {
     'url': Var('chromium_url') + '/catapult.git' + '@' + '21cd5a309631b78d6deedc1ca56ca83ead9fbadd',
     'condition': 'checkout_android',
@@ -74,7 +66,6 @@ deps = {
 
 recursedeps = [
   'v8/buildtools',
-  'v8/third_party/android_tools',
 ]
 
 include_rules = [
