diff --git a/DEPS b/DEPS
index da388a6d36..3df48e5258 100644
--- a/DEPS
+++ b/DEPS
@@ -24,14 +24,6 @@ deps = {
     Var('chromium_url') + '/chromium/buildtools.git' + '@' + '2dff9c9c74e9d732e6fe57c84ef7fd044cc45d96',
   'v8/base/trace_event/common':
     Var('chromium_url') + '/chromium/src/base/trace_event/common.git' + '@' + '211b3ed9d0481b4caddbee1322321b86a483ca1f',
-  'v8/third_party/android_ndk': {
-    'url': Var('chromium_url') + '/android_ndk.git' + '@' + '4e2cea441bfd43f0863d14f57b1e1844260b9884',
-    'condition': 'checkout_android',
-  },
-  'v8/third_party/android_tools': {
-    'url': Var('chromium_url') + '/android_tools.git' + '@' + '130499e25286f4d56acafa252fee09f3cc595c49',
-    'condition': 'checkout_android',
-  },
   'v8/third_party/catapult': {
     'url': Var('chromium_url') + '/catapult.git' + '@' + 'c80faf456d4ca70b4782073e999f90a47c0598c9',
     'condition': 'checkout_android',
@@ -80,7 +72,6 @@ deps = {
 
 recursedeps = [
   'v8/buildtools',
-  'v8/third_party/android_tools',
 ]
 
 include_rules = [
