diff --git a/BUILD.gn b/BUILD.gn
index 10fee26420..60fd08d2a5 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -703,14 +703,7 @@ config("toolchain") {
 # optimize some files even in debug builds to speed up mksnapshot times.
 config("always_optimize") {
   configs = [ ":internal_config" ]
-
-  # TODO(crbug.com/621335) Rework this so that we don't have the confusion
-  # between "optimize_speed" and "optimize_max".
-  if (((is_posix && !is_android) || is_fuchsia) && !using_sanitizer) {
-    configs += [ "//build/config/compiler:optimize_speed" ]
-  } else {
-    configs += [ "//build/config/compiler:optimize_max" ]
-  }
+  configs += [ "//build/config/compiler:optimize_speed" ]
 }
 
 # Configs for code coverage with gcov. Separate configs for cflags and ldflags
