diff --git a/BUILD.gn b/BUILD.gn
index c486f1a1c8..5412b0758c 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -739,13 +739,7 @@ config("toolchain") {
 config("always_optimize") {
   configs = [ ":internal_config" ]
 
-  # TODO(crbug.com/621335) Rework this so that we don't have the confusion
-  # between "optimize_speed" and "optimize_max".
-  if (((is_posix && !is_android) || is_fuchsia) && !using_sanitizer) {
-    configs += [ "//build/config/compiler:optimize_speed" ]
-  } else {
-    configs += [ "//build/config/compiler:optimize_max" ]
-  }
+  configs += [ "//build/config/compiler:optimize_speed" ]
 }
 
 # Configs for code coverage with gcov. Separate configs for cflags and ldflags
diff --git a/gni/v8.gni b/gni/v8.gni
index e55c4cf346..e0c02cd0f6 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -111,13 +111,7 @@ if (is_debug && !v8_optimized_debug) {
 } else {
   v8_remove_configs += [ "//build/config/compiler:default_optimization" ]
 
-  # TODO(crbug.com/621335) Rework this so that we don't have the confusion
-  # between "optimize_speed" and "optimize_max".
-  if (((is_posix && !is_android) || is_fuchsia) && !using_sanitizer) {
-    v8_add_configs += [ "//build/config/compiler:optimize_speed" ]
-  } else {
-    v8_add_configs += [ "//build/config/compiler:optimize_max" ]
-  }
+  v8_add_configs += [ "//build/config/compiler:optimize_speed" ]
 }
 
 if (v8_code_coverage && !is_clang) {
