From 5993d9929792f969c1543b91343bf001eef5085d Mon Sep 17 00:00:00 2001
From: Gary Mathews <gmathews@appcelerator.com>
Date: Mon, 2 Dec 2019 15:41:28 -0800
Subject: [PATCH] toolchain

---
 build/toolchain/mac/BUILD.gn | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/build/toolchain/mac/BUILD.gn b/build/toolchain/mac/BUILD.gn
index c58a13cb10..0d55c3c4df 100644
--- a/build/toolchain/mac/BUILD.gn
+++ b/build/toolchain/mac/BUILD.gn
@@ -520,14 +520,18 @@ mac_toolchain("clang_x64") {
 
 mac_toolchain("clang_x86") {
   toolchain_args = {
-    current_cpu = "x86"
+    current_cpu = "x64"
     current_os = "mac"
+
+    if (defined(v8_current_cpu)) {
+      v8_current_cpu = "x86"
+    }
   }
 }
 
 mac_toolchain("clang_x86_v8_arm") {
   toolchain_args = {
-    current_cpu = "x86"
+    current_cpu = "x64"
     current_os = "mac"
 
     if (defined(v8_current_cpu)) {
@@ -538,7 +542,7 @@ mac_toolchain("clang_x86_v8_arm") {
 
 mac_toolchain("clang_x86_v8_mipsel") {
   toolchain_args = {
-    current_cpu = "x86"
+    current_cpu = "x64"
     current_os = "mac"
 
     if (defined(v8_current_cpu)) {
-- 
2.21.0 (Apple Git-122)

